src/app/portlets/dot-templates/dot-template/dot-template.component.ts

```diff
@@ -0,0 +1,110 @@
+import { DotContainer } from '@shared/models/container/dot-container.model';
+import { Subject } from 'rxjs';
+import { take, takeUntil } from 'rxjs/operators';
+import * as _ from 'lodash';
+import { DotTemplate } from '@portlets/dot-edit-page/shared/models';
+
+@Component({
+    selector: 'dot-template-test',
+    templateUrl: './dot-template.component.html',
+    styleUrls: ['./dot-template.scss']
+})
+export class DotTemplateComponent implements OnInit, OnDestroy {
+    group: FormGroup;
+    editor: any;
+    isDisabled = true;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+    private originalValue: DotTemplate;
+
+    constructor(
+        private fb: FormBuilder,
+        private dotTemplateService: DotTemplatesService,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    ngOnInit(): void {
+        this.group = this.fb.group({
+            title: ['', Validators.required],
+            friendlyName: '',
+            body: ['', Validators.required]
+        });
+
+        this.initialFormValues();
+        this.onFormValueChanges();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+    /**
+     * This method initializes the monaco editor
+     *
+     * @param {*} editor
+     * @memberof DotTemplateComponent
+     */
+    initEditor(editor) {
+        this.editor = editor;
+    }
+
+    /**
+     * This method handles the submit event of the form
+     *
+     * @param {*} event
+     * @memberof DotTemplateComponent
+     */
+    onSubmit(event) {
+        event.preventDefault();
+        this.dotTemplateService
+            .create(this.group.value)
+            .pipe(take(1))
+            .subscribe(() => {
+                if (this.group.valid) {
+                    this.dotRouterService.goToURL('/c/templates');
+                }
+            });
+    }
+    /**
+     * This method handles the change event of the searchable selector and
+     * inserts the container to the editor
+     *
+     * @param {DotContainer} container
+     * @memberof DotTemplateComponent
+     */
+    containerChange(container: DotContainer): void {
+        const selection = this.editor.getSelection();
+
+        const id = this.setContainerId(container);
+
+        const text = `## Container: ${
+            container.name
+        }\n## This is autogenerated code that cannot be changed\n#parseContainer('${id}','${Date.now()}')\n`;
+        const operation = { range: selection, text: text, forceMoveMarkers: true };
+        this.editor.executeEdits('source', [operation]);
+    }
+
+    private initialFormValues() {
+        this.originalValue = this.group.value;
+    }
+
+    private onFormValueChanges() {
+        this.group.valueChanges.pipe(takeUntil(this.destroy$)).subscribe((values) => {
+            this.compareValues(values, this.originalValue);
+        });
+    }
+
+    private setContainerId(container: DotContainer) {
```

Comment:
If the method returns... is not a set is a get.

```suggestion
    private setContainerId(container: DotContainer): string {
```